pkgs/build-support/writers/test.nix

```diff
@@ -192,13 +229,25 @@ let
       fi
     '';
 
+  enabledTest = name: test:
+    let enabled = if builtins.hasAttr name enabledTests
+                    then builtins.getAttr name enabledTests
+                    else false;
+    in
+      lib.optionalString enabled test;
+
 in runCommand "test-writers" {
-  passthru = { inherit writeTest bin simple; };
+  passthru = { inherit writeTest bin simple trivial path; };
   meta.platforms = lib.platforms.all;
 } ''
-  ${lib.concatMapStringsSep "\n" (test: writeTest "success" test.name "${test}/bin/${test.name}") (lib.attrValues bin)}
-  ${lib.concatMapStringsSep "\n" (test: writeTest "success" test.name test) (lib.attrValues simple)}
-  ${lib.concatMapStringsSep "\n" (test: writeTest "success" test.name test) (lib.attrValues path)}
+  ${enabledTest "bin" (lib.concatMapStringsSep "\n" (test: writeTest "success" test.name "${test}/bin/${test.name}") (lib.attrValues bin))}
+  # NOTE: this is a bit of a hack. We don't want to actually run the outputs of these tests, because they are the
+  # _result_ scripts of the writeShellScript, not the checker script. The "test" is run implicitly in that
+  # the mkDerivation's buildCommand will abort if the checkPhase (from writeTextFile) fails.
+  # So: the comment here forces the evaluation but doesn't actually call the scripts.
+  #${enabledTest "trivial" (lib.concatStringsSep " " (lib.attrValues trivial))}
```

Comment:
Why not run them? Just make something that outputs `success` and uses the crazy globbing stuff in a way that doesn't print anything to `stdout`.